onedal/ensemble/tests/test_random_forest.py

```diff
@@ -51,7 +51,7 @@ def test_rf_regression(queue):
 
 
 @pytest.mark.parametrize("queue", get_queues("gpu"))
-def test_rf_regression(queue):
+def test_rf_regression_gpu(queue):
```

Comment:
Bad design.
If gpu device is not available, you will get empty list for queue. In this case your getting undef behavior at least.
Even passing this test case when gpu device is not available is meaningful. 

Please do not separate device checks between test suits